### PR TITLE
Add release notes

### PR DESCRIPTION
**Pull request by 9999years**
I've forgotten to add release notes for some of my PRs, and fixes to some issues like #7245 have gone unremarked in the release notes. This PR adds them to the `rl-next` directory (so they'll get seen by people who watch release notes) and amends the historical release notes where applicable (so they'll get seen by people who look at historical release notes).

### CODE REVIEW

File: doc/manual/src/release-notes/rl-2.20.md [hunk 1]
```diff
@@ -167,3 +167,55 @@
...
+  ```
+
+- Flake operations like `nix develop` will no longer fail when run in a Git
+  repository where the `flake.lock` file is `.gitignore`d
+  [#8854](https://github.com/NixOS/nix/issues/8854)
+  [#9324](https://github.com/NixOS/nix/pull/9324)
+
+- Nix commands will now respect Ctrl-C
+  [#7154](https://github.com/NixOS/nix/issues/7145)
+  [#6695](https://github.com/NixOS/nix/pull/6695)
```
Comment (edolstra) [inline doc/manual/src/release-notes/rl-2.20.md:196]: This looks unrelated to Ctrl-C?

**Comment by edolstra** [comment]
Thanks! Release notes are very welcome.

However, we do want the release notes to be fairly terse, since we want people to read all of them. So we generally don't include release notes for bug fixes (e.g. the Ctrl-C issue) or error message tweaks (like removing some newlines from the output).

File: doc/manual/rl-next/flake-add-git-ignored-files.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+       … while updating the lock file of flake 'git+file:///Users/wiggles/.dotfiles?dir=config/home-mangler&ref=refs/heads/main&rev=f4d806103ef76becbc220c8e3d09bf6ac1e78bb8'
+
+       error: program 'git' failed with exit code 1
+```
+
+Now, Nix will detect that the file is `.gitignore`d and skip adding it, making
+commands like `nix develop` succeed.
+
+This was actually released in Nix 2.20, but wasn't added to the release notes
+so we're announcing it here. The historical release notes have been updated as well.
```
Comment (edolstra) [inline doc/manual/rl-next/flake-add-git-ignored-files.md:28]: This will cause this release notes entry to be duplicated for 2.20 and 2.21. I don't think we want that. So let's only include this in `rl-2.20.md`.

File: doc/manual/src/release-notes/rl-2.20.md (hunk 1 above)
Comment (9999years) [reply doc/manual/src/release-notes/rl-2.20.md]: Good catch, I'd gotten the numbers wrong.

**Comment by 9999years** [comment]
> we generally don't include release notes for bug fixes (e.g. the Ctrl-C issue)

I really think we should — that bug consistently annoys me at work and I was thrilled to see it fixed!

**Comment by 9999years** [comment]
> we generally don't include release notes for ... error message tweaks (like removing some newlines from the output).

OK, removed that one.

File: doc/manual/rl-next/flake-add-git-ignored-files.md (hunk 2 above)
Comment (fricklerhandwerk) [reply doc/manual/rl-next/flake-add-git-ignored-files.md]: Yep, I think it's fine to backport release notes manually for completeness though. We don't have to make a point release just for those.

**Review by thufschmitt — APPROVED** [review]
